pkg/istio-agent/agent.go

```diff
@@ -207,14 +207,14 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, agentOpts *AgentOptions, sopts *sec
 	}
 }
 
-// EnvoyDisabled if true inidcates calling Run will not run and wait for Envoy.
+// EnvoyDisabled if true indicates calling Run will not run and wait for Envoy.
 func (a *Agent) EnvoyDisabled() bool {
 	return a.envoyOpts.TestOnly || a.cfg.DisableEnvoy
 }
 
 // WaitForSigterm if true indicates calling Run will block until SIGKILL is received.
 func (a *Agent) WaitForSigterm() bool {
-	return a.EnvoyDisabled() && !a.envoyOpts.TestOnly
+	return a.cfg.DisableEnvoy
```

Comment:
why is this changed?

Reply:
sorry, that's my bad, i consider it too simplely